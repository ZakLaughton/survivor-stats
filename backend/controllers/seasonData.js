const getSeasonData = async (req, res, db) => {
  let response = {
    season: req.query.season,
    tribes: [],
    episodes: [],
    preseasonStats: []
  };

  const season = req.query.season;

  response.tribes = await db
    .select('name', 'tribe_color')
    .from('tribes')
    .where('season', '=', Number(season))
    .catch(console.log);
  const seasonEpisodes = await db
    .select('*')
    .from('episodes')
    .where('id', 'like', `s${season}e%`)
    .catch(console.log);
  const seasonCastaways = await db
    .select('*')
    .from('season_castaway_mapping')
    .where('season_no', '=', Number(season))
    .join('castaways', 'castaways.full_name', 'season_castaway_mapping.name')
    .catch(console.log);
  const seasonTribeChanges = await db
    .select('castaway', 'field_value', 'start_episode', 'boot_order', 'details')
    .from('updates')
    .where('start_episode', 'like', `s${season}%`)
    .where('field', '=', 'tribe');
  const seasonAdvantageChanges = await db
    .select(
      'castaway',
      'field_value',
      'start_episode',
      'end_episode',
      'details'
    )
    .from('updates')
    .where('start_episode', 'like', `s${season}%`)
    .where('field', '=', 'advantage');
  const castawayProfiles = await db('castaway_season_profiles').where(
    'season',
    '=',
    Number(season)
  );

  const preseasonStats = await db
    .select('stat', 'value', 'line')
    .from('preseason_stats')
    .where('season', '=', Number(season))
    .andWhere('display', '=', 'true');

  const tribalCouncils = await db('tribal_councils').where(
    'episode',
    'like',
    `s${season}%`
  );

  const tribalPlays = await db('tribal_plays')
    .join(
      'tribal_councils',
      'tribal_plays.tribal_council',
      '=',
      'tribal_councils.id'
    )
    .where('tribal_councils.episode', 'like', `s${season}%`);

  const castawayDataByEpisode = seasonEpisodes.map(episode => {
    // set up empty skeleton for each episode object
    const episodeObj = {
      id: null,
      castaways: [],
      tribalCouncils: []
    };

    episodeObj.active = episode.active;
    episodeObj.id = `s${season}e${episode.id.slice(-2)}`;
    episodeObj.castaways = seasonCastaways
      .map(castaway => ({
        name: castaway.name,
        nickname: castaway.nickname,
        currentBoot: false,
        juryMember: false,
        bootOrder: null
      }))
      .sort((a, b) => (a.name < b.name ? -1 : 1));

    // Populate tribe data for each castaway
    episodeObj.castaways = episodeObj.castaways.map(castaway => {
      const updatedCastaway = castaway;
      const currentChanges = seasonTribeChanges
        .filter(change => change.start_episode <= episodeObj.id) // filter later episode data
        .filter(currentChange => currentChange.castaway === castaway.name);

      // find the episode(s) with the highest change.start_episode
      const latestChangeEpisode = currentChanges.reduce((prev, current) =>
        prev.start_episode > current.start_episode ? prev : current
      ).start_episode;

      // Get changes from the most recent episode
      const latestChanges = currentChanges.filter(
        change => change.start_episode === latestChangeEpisode
      );
      // Get previous changes for former tribes
      const previousChanges = currentChanges
        .filter(change => change.start_episode != latestChangeEpisode)
        .sort((a, b) => (a.start_episode < b.start_episode ? -1 : 1));
      // To retrieve last tribe for booted contestants
      const latestNonOutChange = currentChanges
        .filter(change => change.field_value != 'out')
        .reduce((prev, current) =>
          prev.start_episode > current.start_episode ? prev : current
        );
      updatedCastaway.formerTribes = previousChanges.map(
        change => change.field_value
      );

      // Handle booted contestants
      if (latestChanges.some(change => change.field_value === 'out')) {
        // Handle contestants booted in THIS episode
        if (latestChangeEpisode === episodeObj.id) {
          updatedCastaway.currentBoot = true;
          updatedCastaway.tribe = latestNonOutChange.field_value;
        } else {
          updatedCastaway.tribe = 'out';
        }
        if (latestChanges.some(change => change.details === 'jury')) {
          updatedCastaway.juryMember = true;
        }
        updatedCastaway.bootOrder = latestChanges.find(
          change => change.field_value === 'out'
        ).boot_order;
      } else {
        updatedCastaway.tribe = latestChanges[0].field_value;
      }

      return updatedCastaway;
    });

    // Populate advantage data for each castaway
    episodeObj.castaways = episodeObj.castaways.map(castaway => {
      const updatedCastaway = castaway;
      const currentChanges = seasonAdvantageChanges
        .filter(change => change.start_episode <= episodeObj.id) // filter later advantages
        .filter(
          change => !change.end_episode || change.end_episode > episodeObj.id
        ) // filter past advantages
        .filter(currentChange => currentChange.castaway === castaway.name); // get only changes for castaway

      updatedCastaway.advantages = currentChanges.map(change => {
        return { item: change.field_value, details: change.details };
      });

      if (castawayProfiles.length > 0) {
        wikiUrlPath = castawayProfiles.find(
          castawayProfile => castawayProfile.castaway === castaway.name
        ).wiki_url;
        updatedCastaway.wikiUrl = `https://survivor.fandom.com/wiki/${wikiUrlPath}`;
      }

      return updatedCastaway;
    });

    // Populate tribal council data for each episode
    episodeObj.tribalCouncils = tribalCouncils
      .filter(tribalCouncil => tribalCouncil.episode === episodeObj.id)
      .map(episodeTribalCouncil => {
        const tribalCouncilObject = {
          id: episodeTribalCouncil.id,
          tribalNumber: episodeTribalCouncil.tribal_number,
          tribe: episodeTribalCouncil.tribe,
          finalTribal: episodeTribalCouncil.final_tribal,
          fireMakingTribal: episodeTribalCouncil.fire_making_tribal,
          castawayVotedFor: episodeTribalCouncil.castaway_voted_out,
          notes: episodeTribalCouncil.notes,
          day: episodeTribalCouncil.day_number,
          vote_rounds: []
        };

        // populate vote rounds
        for (let i = 1; i <= episodeTribalCouncil.vote_rounds; i++) {
          let voteRound = {
            round_no: i,
            votes: [],
            advantages: []
          };

          // get only plays from this vote round
          currentPlays = tribalPlays.filter(
            tribalPlay =>
              tribalPlay.tribal_council === tribalCouncilObject.id &&
              tribalPlay.vote_no === i
          );
          // add votes to round
          voteRound.votes = currentPlays
            .filter(tribalPlay => tribalPlay.item_played === 'vote')
            .map(vote => {
              return {
                playedBy: vote.played_by,
                playedOn: vote.played_on
              };
            });

          // add advantage plays to round
          voteRound.advantages = currentPlays
            .filter(tribalPlay => tribalPlay.item_played !== 'vote')
            .map(advantage => {
              return {
                advantage: advantage.item_played,
                playedBy: advantage.played_by,
                playedOn: advantage.played_on
              };
            });

          tribalCouncilObject.vote_rounds.push(voteRound);
        }

        // populate advantage plays

        return tribalCouncilObject;
      });

    // Populate played vote and advantage data for each tribal council

    episodeObj.tribalCouncils = episodeObj.tribalCouncils.map(
      tribal_council => {
        let updatedTribalCouncil = tribal_council;
        return updatedTribalCouncil;
      }
    );

    return episodeObj;
  });

  response.episodes = castawayDataByEpisode;
  response.preseasonStats = preseasonStats;
  res.json(response);
};

module.exports = {
  getSeasonData
};
