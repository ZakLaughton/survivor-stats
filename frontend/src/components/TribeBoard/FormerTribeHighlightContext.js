import React from 'react';

export const FormerTribeHighlightContext = React.createContext();
