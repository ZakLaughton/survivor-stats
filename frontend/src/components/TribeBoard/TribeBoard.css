.active-tribes {
  display: flex;
  min-height: 100vh;
}

.clear-footer {
    clear: both; height: 80px;
}