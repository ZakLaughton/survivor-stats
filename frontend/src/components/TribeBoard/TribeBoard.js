import React from 'react';
import Tribe from '../Tribe/Tribe';
import VotedOutPanel from '../VotedOutPanel/VotedOutPanel';
import './TribeBoard.css'

class TribeBoard extends React.Component {
  state = {
    activeTribes: [],
    episodeData: {},
    formerTribeHighlight: {
      active: false,
      tribeName: null
    }
  }

  setActiveTribes = (seasonData, episodeId) => {
    const {tribes} = seasonData;
    const episodeData = seasonData.episodes.find((episode) => {
       return episode.id === episodeId;
    })
    const activeTribes = tribes.filter((tribe) => {
      return episodeData.castaways
        // Don't show current boots (to be removed in future)
        .filter(castaway => castaway.currentBoot === false)
        .some(castaway => castaway.tribe.replace(/\d| /g, '') === tribe.name);
    })

    this.setState({activeTribes})
  }

  setEpisodeData = (seasonData, episodeId) => {
    const episodeData = seasonData.episodes.find(episode => episode.id === episodeId);
    this.setState({episodeData});
  }

  setFormerTribeHighlight = (formerTribe) => {
    const updatedTribeHighlight = this.state.formerTribeHighlight;
    updatedTribeHighlight.active = true;
    updatedTribeHighlight.tribeName = formerTribe;
    this.setState({formerTribeHighlight: updatedTribeHighlight});
  }

  removeFormerTribeHighlight = () => {
    console.log('removing...')
    const updatedTribeHighlight = this.state.formerTribeHighlight;
    updatedTribeHighlight.active = false;
    updatedTribeHighlight.tribeName = null;
    this.setState({formerTribeHighlight: updatedTribeHighlight});
  }

  componentDidMount() {
    const { seasonData, episodeId } = this.props;
    this.setActiveTribes(seasonData, episodeId)
    this.setEpisodeData(seasonData, episodeId)
  }
  
  componentWillReceiveProps(nextProps) {
    const { seasonData, episodeId } = nextProps;
    if (seasonData.episodes) {
      this.setActiveTribes(seasonData, episodeId)
      this.setEpisodeData(seasonData, episodeId)
    } 
  }

  render() {
    const {activeTribes, episodeData, formerTribeHighlight} = this.state;
    const {tribeData} = this.props;
    const {setFormerTribeHighlight, removeFormerTribeHighlight} = this;
    return(
      <main>
        <div className="active-tribes">
          {activeTribes.length > 0 &&
            activeTribes.map(tribe => (
              <Tribe
                key={tribe.name}
                tribe={tribe}
                episodeData={episodeData}
                tribeData={tribeData}
                formerTribeHighlight={formerTribeHighlight}
                setFormerTribeHighlight={setFormerTribeHighlight}
                removeFormerTribeHighlight={removeFormerTribeHighlight}  
              />
            ))
          }
          {activeTribes.length === 0 && 'loading...'}
        </div>
        <VotedOutPanel episodeData={episodeData}/>
      </main>
    )
  }
}

export default TribeBoard;