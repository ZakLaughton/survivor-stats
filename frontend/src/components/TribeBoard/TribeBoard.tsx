import styled, { createGlobalStyle } from "styled-components";
import React, { useState } from "react";
import VotedOutPanel from "../VotedOutPanel/VotedOutPanel";
import { FormerTribeHighlightContext } from "./FormerTribeHighlightContext";
import {ActiveSeasonData, Tribe as TribeType, Episode} from "../../types";
import Tribe from "../Tribe/Tribe"

const castawayCardSizeSm = `110px`;

const GlobalStyle = createGlobalStyle`
  @media only screen and (min-width: 761px) and (max-width: 900px) {
    .tribe-count-4.active-tribes {
      flex-flow: row wrap !important;
    }

    .tribe-count-4 .extinction-island {
      flex: 0 1 100% !important;
    }

    .tribe-count-4 .extinction-island h1 {
      margin-bottom: 0;
    }
  }
`;

// @ts-ignore
export const FormerTribeHighlightProvider = ({ children }) => {
  const [highlightedFormerTribe, setFormerTribeHighlight] = useState({
    tribeName: ``,
    color: `blue`,
  });
  return (
    <FormerTribeHighlightContext.Provider
      value={{
        highlightedFormerTribe,
        // @ts-ignore
        updateTribeHighlight: (tribeName: string) => setFormerTribeHighlight(tribeName),
      }}
    >
      {children}
    </FormerTribeHighlightContext.Provider>
  );
};

interface TribeBoardProps {
  tribeData: TribeType;
  activeSeasonData: ActiveSeasonData;
  episodeId: string;
  seasonNum: number;
}

const TribeBoard = ({
  tribeData, activeSeasonData, episodeId, seasonNum,
}: TribeBoardProps) => {
  let episodeData: Episode | undefined;
  if (activeSeasonData && activeSeasonData.episodes && activeSeasonData.episodes.find(episode => episode.id === episodeId)) {
    episodeData = activeSeasonData.episodes.find(episode => episode.id === episodeId)
  }

  const activeTribes: TribeType[] = activeSeasonData && activeSeasonData.tribes && episodeData && episodeData.castaways
  // @ts-ignore episodeData object is possibly 'undefined'
  ? activeSeasonData.tribes.filter(tribe => episodeData.castaways
    // Don't show current boots (to be removed in future)
      .filter(castaway => castaway.currentBoot === false)
      .some(castaway => castaway.tribe.replace(/ \d/g, ``) === tribe.name))
    : [];

  return (
    <FormerTribeHighlightProvider>
      <article>
        <GlobalStyle />
        {/*
        // @ts-ignore */}
        <ActiveTribes activeTribes={activeTribes} className={`tribe-count-${activeTribes.length}`}>
          {activeTribes.length > 0
            && activeTribes
              .filter(tribe => tribe.name !== `Extinction Island`)
              .map((tribe: TribeType) => (
                <Tribe
                  key={tribe.name}
                  tribe={tribe}
                  episodeData={episodeData}
                  tribeData={tribeData}
                  seasonNum={seasonNum}
                />
              ))}
          {activeTribes.length > 0
            && activeTribes
              .filter(tribe => tribe.name === `Extinction Island`)
              .map(tribe => (
                <Tribe
                  key={tribe.name}
                  tribe={tribe}
                  episodeData={episodeData}
                  tribeData={tribeData}
                  seasonNum={seasonNum}
                />
              ))}
          {activeTribes.length === 0 && `loading...`}
        </ActiveTribes>
        {/*
        // @ts-ignore */}
        <VotedOutPanel episodeData={episodeData} tribeData={tribeData} seasonNum={seasonNum} />
      </article>
    </FormerTribeHighlightProvider>
  );
};

const ActiveTribes = styled.div`
  display: flex;
  flex-direction: row;
  flex: 1 1 auto;

  ${(props) => {
    // @ts-ignore - Property 'activeTribes' does not exist on type 'ThemedStyledProps...
    if (props.activeTribes.length === 2) {
      return `@media only screen and (max-width: 750px) {flex-direction: column;}`;
    }
    // @ts-ignore - Property 'activeTribes' does not exist on type 'ThemedStyledProps...
    if (props.activeTribes.length === 3) {
      return `@media only screen and (max-width: 766px) {flex-direction: column;}`;
    }
    // @ts-ignore - Property 'activeTribes' does not exist on type 'ThemedStyledProps...
    if (props.activeTribes.length === 4) {
      return `@media only screen and (max-width: 761px) {flex-direction: column;}`;
    }
    return ``;
  }}
`;

export default TribeBoard;
