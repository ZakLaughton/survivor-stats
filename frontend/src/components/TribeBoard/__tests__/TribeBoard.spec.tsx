import React from 'react';
import { seasonData } from '../../../__fixtures__/seasonData';
import { render } from '@testing-library/react';
import { TribeBoard, TribeBoardProps } from '../TribeBoard';

jest.mock('cloudinary-react')
const defaultProps: TribeBoardProps = {
  activeEpisodeNumber: 3,
  activeSeasonData: seasonData
}

it('renders all subcomponents', () => {
  const { getByTestId, getAllByTestId, container } = render(<TribeBoard {...defaultProps}/>);
  expect(getAllByTestId('tribe')).toHaveLength(3);
  expect(getByTestId('votedOutPanel')).toBeTruthy();
  expect(container).toMatchSnapshot();
});
