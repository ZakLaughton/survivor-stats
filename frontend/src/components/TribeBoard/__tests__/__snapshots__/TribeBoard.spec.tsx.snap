// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all subcomponents 1`] = `
<div>
  <section
    class="sc-bZQynM jbYBQB"
  >
    <h1
      class="sc-gzVnrw ckkwht"
    >
      Standings
    </h1>
    <div
      class="sc-chPdSV aaTih tribe-count-3"
    >
      <section
        class="sc-jzJRlG faDmHW tribe pa2 fl vuku"
        data-testid="tribe"
      >
        <h2
          class="sc-cSHVUG fKkrPJ"
        >
          Vuku
        </h2>
        <div
          class="sc-kAzzGY cxGlqK"
        >
          <div
            class="sc-gZMcBi kyBWZI castaway-card grow relative former-goliath"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Alec_Merlino"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Alec
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: purple;"
              >
                <span
                  class="tooltip-text"
                >
                  Former Goliath
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
          <div
            class="sc-gZMcBi kyBWZI castaway-card grow relative former-david"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Carl_Boudreaux"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Carl
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: orange;"
              >
                <span
                  class="tooltip-text"
                >
                  Former David
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            >
              <div>
                <i
                  class="sc-dnqmqq eWsBgn tooltip advantage fas fa-ban"
                >
                  <span
                    class="animated fadeIn tooltip-text"
                  >
                    idol nullifier
                  </span>
                </i>
              </div>
            </div>
          </div>
          <div
            class="sc-gZMcBi kyBWZI castaway-card grow relative former-david"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Davie_Rickenbacker"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Davie
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: orange;"
              >
                <span
                  class="tooltip-text"
                >
                  Former David
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            >
              <div>
                <div
                  class="tooltip advantage"
                >
                  <img
                    alt="Immunity Idol"
                    src="immunity_idol.png"
                  />
                  <span
                    class="animated fadeIn tooltip-text"
                  >
                    Immunity Idol
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-gZMcBi kyBWZI castaway-card grow relative former-david"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Elizabeth_Olson"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Elizabeth
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: orange;"
              >
                <span
                  class="tooltip-text"
                >
                  Former David
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
          <div
            class="sc-gZMcBi kyBWZI castaway-card grow relative former-goliath"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Kara_Kay"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Kara
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: purple;"
              >
                <span
                  class="tooltip-text"
                >
                  Former Goliath
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
        </div>
      </section>
      <section
        class="sc-jzJRlG faDmHW tribe pa2 fl jabeni"
        data-testid="tribe"
      >
        <h2
          class="sc-cSHVUG fKkrPJ"
        >
          Jabeni
        </h2>
        <div
          class="sc-kAzzGY cxGlqK"
        >
          <div
            class="sc-gZMcBi iQHPme castaway-card grow relative former-goliath"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Angelina_Keeley"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Angelina
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: purple;"
              >
                <span
                  class="tooltip-text"
                >
                  Former Goliath
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
          <div
            class="sc-gZMcBi iQHPme castaway-card grow relative former-david"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Lyrsa_Torres"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Lyrsa
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: orange;"
              >
                <span
                  class="tooltip-text"
                >
                  Former David
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
          <div
            class="sc-gZMcBi iQHPme castaway-card grow relative former-goliath"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Mike_White"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Mike
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: purple;"
              >
                <span
                  class="tooltip-text"
                >
                  Former Goliath
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
          <div
            class="sc-gZMcBi iQHPme castaway-card grow relative former-goliath"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Natalie_Cole"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Natalie
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: purple;"
              >
                <span
                  class="tooltip-text"
                >
                  Former Goliath
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
          <div
            class="sc-gZMcBi iQHPme castaway-card grow relative former-david"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Nick_Wilson"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Nick
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: orange;"
              >
                <span
                  class="tooltip-text"
                >
                  Former David
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
        </div>
      </section>
      <section
        class="sc-jzJRlG faDmHW tribe pa2 fl tiva"
        data-testid="tribe"
      >
        <h2
          class="sc-cSHVUG fKkrPJ"
        >
          Tiva
        </h2>
        <div
          class="sc-kAzzGY cxGlqK"
        >
          <div
            class="sc-gZMcBi fFFTWl castaway-card grow relative former-goliath"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Alison_Raybould"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Alison
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: purple;"
              >
                <span
                  class="tooltip-text"
                >
                  Former Goliath
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
          <div
            class="sc-gZMcBi fFFTWl castaway-card grow relative former-david"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Christian_Hubicki"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Christian
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: orange;"
              >
                <span
                  class="tooltip-text"
                >
                  Former David
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
          <div
            class="sc-gZMcBi fFFTWl castaway-card grow relative former-goliath"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Dan_Rengering"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Dan
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: purple;"
              >
                <span
                  class="tooltip-text"
                >
                  Former Goliath
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            >
              <div>
                <div
                  class="tooltip advantage"
                >
                  <img
                    alt="Immunity Idol"
                    src="immunity_idol.png"
                  />
                  <span
                    class="animated fadeIn tooltip-text"
                  >
                    Immunity Idol
                  </span>
                </div>
              </div>
            </div>
          </div>
          <div
            class="sc-gZMcBi fFFTWl castaway-card grow relative former-david"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/Gabby_Pascuzzi"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              Gabby
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: orange;"
              >
                <span
                  class="tooltip-text"
                >
                  Former David
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
          <div
            class="sc-gZMcBi fFFTWl castaway-card grow relative former-goliath"
          >
            <a
              class="sc-gqjmRU fwhydA"
              href="https://survivor.fandom.com/wiki/John_Hennigan"
              target="_blank"
            >
              <div
                class="sc-bdVaJa dllXJH"
                size="90"
              />
            </a>
            <div
              class="sc-VigVT fVBOYd card-nameplate"
            >
              John
            </div>
            <div
              class="sc-jTzLTM krOZGW tribe-circle-container"
            >
              <i
                class="fas fa-history"
              />
              <div
                class="sc-iwsKbI dKkpcS tribe-circle tooltip"
                style="background-color: purple;"
              >
                <span
                  class="tooltip-text"
                >
                  Former Goliath
                </span>
              </div>
            </div>
            <div
              class="tribe-advantage-container"
            />
          </div>
        </div>
      </section>
    </div>
    <section
      class="sc-htpNat kyFgmI voted-out-panel animated slideInUp"
      data-testid="votedOutPanel"
    >
      <div
        class="sc-ifAKCX eiYjAL"
      >
        <span
          class="sc-EHOje itjQSb"
        >
          OUT
        </span>
        <div
          class="sc-bdVaJa SitNr"
          size="64"
        />
        <div
          class="sc-bdVaJa SitNr"
          size="64"
        />
        <div
          class="sc-bdVaJa SitNr"
          size="64"
        />
        <div
          class="sc-bdVaJa SitNr"
          size="64"
        />
        <div
          class="sc-bdVaJa SitNr"
          size="64"
        />
      </div>
    </section>
  </section>
</div>
`;
